pilot/docker/Dockerfile.proxyv2

```diff
@@ -41,5 +41,7 @@ ENV ISTIO_META_ISTIO_VERSION $istio_version
 ARG TARGETARCH
 COPY ${TARGETARCH:-amd64}/pilot-agent /usr/local/bin/pilot-agent
 
+USER 1337:1337
```

Comment:
This is probably a breaking change

Reply:
Why? All tests passed.

Reply:
In general, because of the unknown unknowns.

Unfortunately we do not test every way everyone is using this... but I am certain there are people running the "Kubernetes YAML" from https://istio.io/latest/docs/setup/additional-setup/gateway/#deploying-a-gateway before we added the runAsUser section though. This change will switch them from root to 1337. Which, in general, is good - but the same people are also binding to port `80` which requires root,

Reply:
Can't we add an upgrade note about this breaking change? Or do you consider this a so severe breakage that we should avoid at all costs?

Reply:
This seems likely to cause outages when upgrades occur, especially since the gateway admin may not even be the one upgrading Istio - so they may not have even read the release note